tests/pytest_tests/system_tests/test_artifacts/test_wandb_artifacts.py

```diff
@@ -284,6 +285,48 @@ def test_add_new_file():
     }
 
 
+def test_add_changed():
+    artifact = wandb.Artifact(type="dataset", name="my-arty")
+    with artifact.new_file("file1.txt") as f:
+        f.write("hello")
+
+    entry1 = artifact.manifest.entries["file1.txt"]
+    assert entry1.path == "file1.txt"
+    assert Path(entry1.local_path).read_text() == "hello"
+
+    with artifact.new_file("file1.txt") as f:
+        f.write("goodbye")
+
+    entry2 = artifact.manifest.entries["file1.txt"]
+    assert entry1 != entry2
```

Comment:
Nit: we could reuse the same entry, this check isn't necessary. Same on line 303 below.